cli/cli_state.ml

```diff
@@ -13,7 +13,7 @@ let next_display_mode = function
 type notify_v =
   | Disconnected
   | Connected
-  | Notifications
+  | Notification of Xjid.t
```

Comment:
Ack, yes i can take this out. But there's another issue which I didn't yet handle - notify_v indicates the "state" of jackline, but we might get events that don't trigger a change in state, such as receiving a message from a focused contact. (The TODO I left in maybe_notify.)

Before this PR, notification_callback would be fired on every message, because every message unconditionally set the "has notifications" state. However with this PR, we'll need to refine this model a bit, and so I'll need to make another commit on top of this fix.